packages/backend/src/peripherals/database/ForcedWithdrawRepository.ts

```diff
@@ -0,0 +1,368 @@
+import { Hash256, StarkKey, Timestamp } from '@explorer/types'
+import {
+  ForcedWithdrawStatusRow,
+  ForcedWithdrawTransactionRow,
+} from 'knex/types/tables'
+
+import { Logger } from '../../tools/Logger'
+import { compareByHistory } from '../../utils/compareByHistory'
+import { BaseRepository } from './shared/BaseRepository'
+import { Database } from './shared/Database'
+
+export interface ForcedWithdrawTransactionRecord {
+  hash: Hash256
+  starkKey: StarkKey
+  amount: bigint
+  positionId: bigint
+}
+
+export interface ForcedWithdrawTransactionWithHistory
+  extends ForcedWithdrawTransactionRecord {
+  history: ForcedWithdrawStatus[]
+}
+
+export type ForcedWithdrawStatus =
+  | SentStatus
+  | MinedStatus
+  | RevertedStatus
+  | ForgottenStatus
+  | IncludedStatus
```

Comment:
Wouldn't it be better to declare those Statuses before using them? In terms of clean code of course. I don't know if we have a "rule" for this

Reply:
Makes sense I would make it a convention if that's what we do anyways